src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1427,6 +1427,13 @@ protected override void BeginProcessing()
                 this.ThrowTerminatingError(errorRecord);
```

Comment:
@iSazonov, your last commit had 2 failures in `PowerShell-CI-macos`
Select-String.String actions.Should output a string with all matches highlighted when AllMatch is used
```powershell
Cmdlet parameters AllMatches and SimpleMatch are mutually exclusive.
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Select-String.Tests.ps1: line 100
100:                 $result = $testinputone | Select-String -Pattern "l" -AllMatch | Out-String

```
Select-String.String actions.Should throw if -AllMatches parameter is used with -SimpleMatch parameter
```powershell
Expected an exception, with FullyQualifiedErrorId 'CannotSpecifyAllMatchesWithSimpleMatch,Microsoft.PowerShell.Commands.SelectStringCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Select-String.Tests.ps1: line 111
111:             { "1" | Select-String -Pattern "hello" -AllMatches -SimpleMatch } | Should -Throw -ErrorId "CannotSpecifyAllMatchesWithSimpleMatch,Microsoft.PowerShell.Commands.SelectStringCommand"
```